examples/fNIRS/classify_fNIRS.py

```diff
@@ -0,0 +1,364 @@
+
+        Returns
+        -------
+        self : BlockKernels instance
+            The BlockKernels instance.
+        """
+        return self
+
+    def transform(self, X):
+        """Estimate block kernel or covariance matrices with optional shrinkage.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_channels, n_times)
+            Multi-channel time series.
+
+        Returns
+        -------
+        M : ndarray, shape (n_samples, n_channels, n_channels)
+            Block matrices (kernel or covariance matrices).
+        """
+        n_samples, n_channels, n_times = X.shape
+
+        # Determine block sizes
+        if isinstance(self.block_size, int):
+            if n_channels % self.block_size != 0:
+                raise ValueError(
+                    f"n_channels ({n_channels}) must be divisible by block_size ({self.block_size})"
+                )
+            n_blocks = n_channels // self.block_size
+            blocks = [self.block_size] * n_blocks
+        elif isinstance(self.block_size, (list, np.ndarray)):
+            if sum(self.block_size) != n_channels:
+                raise ValueError(
+                    f"Sum of block sizes ({sum(self.block_size)}) must equal n_channels ({n_channels})"
+                )
+            blocks = self.block_size
+            n_blocks = len(blocks)
+        else:
+            raise ValueError("Parameter block_size must be int or list.")
+
+        # Handle metric parameter
+        if isinstance(self.metric, str):
+            metrics = [self.metric] * n_blocks
+        elif isinstance(self.metric, list):
+            if len(self.metric) != n_blocks:
+                raise ValueError(
+                    f"Length of metric list ({len(self.metric)}) must match number of blocks ({n_blocks})"
+                )
+            metrics = self.metric
+        else:
+            raise ValueError("Parameter metric must be a string or a list of strings.")
+
+        # Handle shrinkage parameter
+        if self.shrinkage is None:
+            shrinkages = [None] * n_blocks
+        elif isinstance(self.shrinkage, (float, int)):
+            shrinkages = [self.shrinkage] * n_blocks
+        elif isinstance(self.shrinkage, list):
+            if len(self.shrinkage) != n_blocks:
+                raise ValueError(
+                    f"Length of shrinkage list ({len(self.shrinkage)}) must match number of blocks ({n_blocks})"
+                )
+            shrinkages = self.shrinkage
+        else:
+            raise ValueError("Parameter shrinkage must be None, a float, or a list of floats.")
+
+        # Lists of current supported metrics
+        kernel_metrics = [
+            'linear', 'poly', 'polynomial', 'rbf', 'laplacian', 'sigmoid', 'cosine'
+        ]
+
+        covariance_estimators = [
+            'corr', 'cov', 'hub', 'lwf', 'mcd', 'oas', 'sch', 'scm', 'stu', 'tyl'
+        ]
+
+        M_matrices = []
+
+
+        for i in range(n_samples):
```

Comment:
I am a bit concerned by this for loop. Our pairwise_kernels functions are vectorized. Technically what you do here boils down to concatenating the tangent space features of each block. What you could do is used a sklearn ColumnTransformer where you pass the list of block indices to the ColumnTransformer and then you can apply a pipeline to each subgroup of signals. You could look at the design of https://coffeine-labs.github.io/coffeine/ 

Also have you compared the performance if you don't bother about the blocks? basically you compute the full matrix? which you can then decide to set to zero in the offdiagonal blocks.

Reply:
Thanks for the feedback! I don't know if you're referring to classification performance or computational efficiency in the second part of your comment.

If it's about classification performance: Yes, we have compared the block-based approach to a full matrix approach, and the block version consistently performed better in our tests. We have also tried adding a "background matrix" to the zero elements of the off-diagonal matrix. This allows for even another kernel function for the HbO and HbR interactions but comes of course at the cost of higher dimensionality of the manifold...

If it's about computational efficiency: No, we haven't specifically optimized for that yet, but I agree there's room for improvement in the code. I'll definitely look into using a ColumnTransformer and review the example from the Coffeine project. Thanks for the suggestion, I will implement this in the coming days.

Reply:
@agramfort, I have just a question for you to clarify: You suggest changing the logic for creating the blocks before stacking them to block diagonal matrices for each trial, right? Is the loop over trials still okay? Because this is the way it's implemented in `block_covariances`. Thanks!

Reply:
yes I think you can avoid the 2 double for loop and just do one loop over blocks and then let the vectorized version of pairwise / covariance functions do the rest over samples.